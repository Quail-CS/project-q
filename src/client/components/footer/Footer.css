/* Footer - 4vh + 4vh = 8vh */
.footer {
  color: #666666;
  display: grid;
  grid-template-rows: 1fr 1fr;
}

/* Footer Links */
.footer-links {
  letter-spacing: 1.5px;
  list-style-type: none;
  margin: 0;
  padding: 0;

  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
}

.footer-links li a {
  color: inherit;
  font-size: 0.8rem;
  padding: 4px 20px;
  text-decoration: none;
}

.footer-links a:hover {
  color: #FFFFFF;
}

/* Disclaimer */
.disclaimer {
  display: flex;
  align-items: center;
  justify-content: center;
}

.disclaimer-text {
  font-size: 0.4rem;
  padding: 0 5vw;
  text-align: center;
}