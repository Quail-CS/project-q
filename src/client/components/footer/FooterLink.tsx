/**
 * @module Footer.tsx
 * @description Application Footer Link Presentation Component
 * UNIT TEST COVERAGE - 100%
 */

import * as React from 'react';
import { Link } from 'react-router-dom';

export interface FooterLinkProps {
  linkKey: string,
  link: string,
  text: string,
}

const FooterLink = (props: FooterLinkProps): JSX.Element => {
  const { linkKey, link, text } = props;
  return (
    <li key={linkKey}><Link to={link}>{text}</Link></li>
  )
}

export default FooterLink;
