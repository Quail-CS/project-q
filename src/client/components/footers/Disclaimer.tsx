/**
 * @module Disclaimer.tsx
 * @description Application Disclaimer Presentation Component
 * UNIT TEST COVERAGE - 100%
 */

import * as React from 'react';

const Disclaimer = () => {
  return (
    <div className="disclaimer">
      <span className="disclaimer-text">
        Ethiq® and Vote With Your Wallet® are registered trademarks of Ethiq Inc. 2018. All rights reserved. Ethiq and other Ethiq products or services referenced here in are either trademarks or registered trademarks of Ethiq, Inc.
        <br />
        Other products and/or company names referenced or mentioned on this site and in media published by Us may be trademarks of their respective owners. We have no affiliation with these companies or their owners, and our services are not endorsed by them.
      </span>
    </div>
  );
};

export default Disclaimer;
