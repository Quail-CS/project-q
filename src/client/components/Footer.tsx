/**
 * @module Footer.tsx
 * @description Application Footer Presentation Component
 */

import * as React from 'react';
import { Link } from 'react-router-dom';
import Disclaimer from './Disclaimer';

const Footer = () => {
  return (
    <div className="footer">
      <ul className="footer-links">
        <li key="privacy-policy"><Link to='/terms/privacy'>PRIVACY POLICY</Link></li>
        <li key="terms-of-service"><Link to='/terms/service'>TERMS AND CONDITIONS</Link></li>
      </ul>
      <Disclaimer />
    </div>
  );
};

export default Footer;
