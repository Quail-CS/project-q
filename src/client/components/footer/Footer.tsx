/**
 * @module Footer.tsx
 * @description Application Footer Presentation Component
 * UNIT TEST COVERAGE - 100%
 */

import * as React from 'react';
import Disclaimer from './Disclaimer';
import FooterLink from './FooterLink';

import './Footer.css';

// TODO: REFACTOR
const footerLinkArray: JSX.Element[] = [];
export const footerLinkData: string[][] = [
  ['privacy-policy', '/terms/privacy', 'PRIVACY POLICY'],
  ['terms-of-service', '/terms/service', 'TERMS AND CONDITIONS'],
  ['frequently-asked-questions', '/terms/faq', 'FAQ'],
];
footerLinkData.forEach(data => footerLinkArray.push(<FooterLink key={data[0]} linkKey={data[0]} link={data[1]} text={data[2]} />))

const Footer = () => {
  return (
    <div className="footer">
      <ul className="footer-links">
        {footerLinkArray}
      </ul>
      <Disclaimer />
    </div>
  );
};

export default Footer;
