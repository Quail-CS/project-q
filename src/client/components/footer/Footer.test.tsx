import * as React from 'react';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as Renderer from 'react-test-renderer';

import Disclaimer from './Disclaimer'
import Footer from './Footer';
import FooterLink from './FooterLink';
import { MemoryRouter } from 'react-router';

Enzyme.configure({ adapter: new Adapter() });

describe('Footer Component: ', () => {
  it('matches the snapshot', () => {
    const tree: Renderer.ReactTestRendererJSON = Renderer.create(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    ).toJSON() as Renderer.ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
  })
  it('renders a <div> with class "footer"', () => {
    const footerWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Footer />);
    expect(footerWrapper.hasClass('footer')).toEqual(true);
    expect(footerWrapper.type()).toEqual('div');
  })
  it('contains one <ul> with class "footer-links" and one <Disclaimer>', () => {
    const footerWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Footer />);
    expect(footerWrapper.find('ul')).toHaveLength(1);
    expect(footerWrapper.find('ul').hasClass('footer-links')).toEqual(true);
    expect(footerWrapper.find('ul')).toHaveLength(1);
    expect(footerWrapper.find(Disclaimer)).toHaveLength(1);
  })
  it('contains one <ul> with class "footer-links" and one <Disclaimer>', () => {
    const footerWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Footer />);
    expect(footerWrapper.find('ul')).toHaveLength(1);
    expect(footerWrapper.find('ul').hasClass('footer-links')).toEqual(true);
    expect(footerWrapper.find('ul')).toHaveLength(1);
    expect(footerWrapper.find(Disclaimer)).toHaveLength(1);
  })
  it('ul contains three <FooterLink> componenets', () => {
    const footerWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Footer />);
    expect(footerWrapper.find('ul').find(FooterLink)).toHaveLength(3);
  })
});
