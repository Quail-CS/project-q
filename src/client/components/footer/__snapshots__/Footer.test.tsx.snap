// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component:  matches the snapshot 1`] = `
<div
  className="footer"
>
  <ul
    className="footer-links"
  >
    <li>
      <a
        href="/terms/faq"
        onClick={[Function]}
      >
        FAQ
      </a>
    </li>
    <li>
      <a
        href="/terms/service"
        onClick={[Function]}
      >
        TERMS AND CONDITIONS
      </a>
    </li>
    <li>
      <a
        href="/terms/privacy"
        onClick={[Function]}
      >
        PRIVACY POLICY
      </a>
    </li>
  </ul>
  <div
    className="disclaimer"
  >
    <span
      className="disclaimer-text"
    >
      Ethiq® and Vote With Your Wallet® are registered trademarks of Ethiq Inc. 2018. All rights reserved. Ethiq and other Ethiq products or services referenced here in are either trademarks or registered trademarks of Ethiq, Inc.
      <br />
      Other products and/or company names referenced or mentioned on this site and in media published by Us may be trademarks of their respective owners. We have no affiliation with these companies or their owners, and our services are not endorsed by them.
    </span>
  </div>
</div>
`;
